internal/pkg/es/result.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package es
+
+import "encoding/json"
+
+// Error
+type ErrorT struct {
```

Comment:
this is moved out from bulk package, since it can be shared for both bulk and non-bulk queries